### ISSUE TITLE
Support Path objects in from_zarr

### ISSUE THREAD

**Issue by jamestwebber** [dask/dask#8168]
Related to #8006 but the opposite side of the coin: while one can save `to_zarr`, it is still not possible to give a `pathlib.Path` object to `dask.array.from_zarr`, although this is possible with`zarr.open`.

**Comment by clbarnes** [comment]
Yes, this seems to be because fsspec does not support `Path`s and `from_zarr` uses fsspec directly where `to_zarr` uses `zarr.open`. On the zarr side, someone passing fsspec mappers around shouldn't be using `Path`s, so it's probably best not to do this `Path` -> `str` conversion in zarr, but here as the fsspec mapper is an internal implementation detail it's probably fair to add another conditional like `if isinstance(url, os.PathLike): url = os.fspath(url)`.
